### PR TITLE
Enhance the DHT Test Plan

### PR DESCRIPTION
**Pull request by daviddias**

### CODE REVIEW

File: plans/dht/README.md [hunk 1]
```diff
@@ -72,3 +79,23 @@ IPFS can safely rely on the latest DHT upgrades by running go-libp2p DHT tests d
     - Nodes ran 5 random-walk queries to populate their Routing Tables
   - **Act I**
     - Each node calls Provide for `i-provides` until it reaches a total of `n-provides`
+
+### `Test:` Provider record saturation
```
Comment (daviddias) [inline plans/dht/README.md]: This plan will benefit from having a controlled network (i.e. position the nodes in known locations in the network). For that, I'll have to make sure I've PeerIds that are evenly distributed over the hash address.

Comment (daviddias) [reply plans/dht/README.md]: Note to self: use https://github.com/libp2p/go-libp2p-kbucket/blob/master/bucket_prefixmap.go for that

File: plans/dht/README.md [hunk 2]
```diff
@@ -72,3 +79,23 @@ IPFS can safely rely on the latest DHT upgrades by running go-libp2p DHT tests d
...
+
+> The libp2p DHT activates Load Balancing techniques once it understand that a node is owning a popular segment of the address space and with that, storing a larger amount of records that what it can handle. This test is designed to verify that libp2p is actively mitigating that by distributing that load with other nodes.
+
+We need to test:
+- Records are replicated to more nodes in the edges of the N closest nodes, once those nodes store a large amount of records (beyond what their computing resources enable)
+- Same as above, but in the inverse direction, that is: records are not replicated to more nodes, once the load is reduced
+- Records are replicated to more nodes in the edges of the N closest nodes, once those records are heavily requested (beyond what their bandwidth enables the N closest nodes to serve them)
+- Same as above, but in the inverse direction, that is: records are not replicated to more nodes, once the load is reduced
+
+### `Test:` Switch modes (client, full)
```
Comment (daviddias) [inline plans/dht/README.md]: This test will require simulated NATs. Will add a `blocked` until that is achievable

**Comment by raulk** [comment]
@daviddias actually, this is a WIP PR. Do you want it merged?

**Comment by daviddias** [comment]
@raulk perhaps no necessary given more recent work from @aschmahmann on testing the DHT? What this PR adds is just a series of test cases based on a old libp2p changelog. Repurpose it if you find it useful in anyway, if not, feel welcome to drop it.

**Comment by raulk** [comment]
The DHT test plan now lives in https://github.com/libp2p/test-plans, and ownership was transferred to the IPFS Content Routing WG in January. This PR is no longer relevant.

**Pull request closed without merging**